test/AutoDiff/superset_adjoint.swift

```diff
@@ -41,30 +41,34 @@ SupersetVJPTests.test("SubsetOfSubset") {
   expectEqual(0, gradient(at: 0, in: { x in foo(x, 0, 0) }))
 }
 
+SupersetVJPTests.test("ApplySubset") {
+  // TF-914
+  @differentiable(wrt: x)
+  func foo<T: Differentiable>(_ x: T, _ y: T, apply: @differentiable (T, T) -> T) -> T {
+    return apply(x, y)
+  }
+  expectEqual(1, gradient(at: Float(0)) { x in foo(x, 0) { $0 + $1 } })
+}
+
 // FIXME: The expression `(+) as @differentiable (Float, @nondiff Float) -> Float)`
 // forms a curry thunk of `Float.+` before conversion to @differentiable, and AD
 // doesn't know how to differentiate the curry thunk, so it produces a
 // "function is not differentiable" error.
-// FIXME: Propagate wrt indices correctly so that this actually takes the
```

Comment:
This was referring to lowering parameter indices to `@nondiff` parameters, which was done long ago.